### ISSUE TITLE
No code found in instance when there is a code

### ISSUE THREAD

**Issue by ewoutkramer** [FirelyTeam/fhir-net-api#632]
See https://chat.fhir.org/#narrow/stream/4-implementers/subject/value.20or.20child.20or.20Null.3F

Validating this will give an error on `Bundle.entry[0].resource[0].dosageInstruction[1].route[0]`:
Code Invalid : No code found in instance.

``` xml
<Bundle xmlns="http://hl7.org/fhir">
<id value="d3e1"/>
<type value="searchset"/>
<total value="1"/>
<entry>
<fullUrl value="medbeh_VMA_001_MA02--1.2.3.999"/>
<resource>
<MedicationRequest>
<meta>
<profile value="http://nictiz.nl/fhir/StructureDefinition/zib-MedicationAgreement"/>
</meta>
<!--voorschrijver-->
<contained>
<Practitioner>
<id value="d3e58"/>
<!--zorgverlener_identificatie_nummer-->
<identifier>
<system value="http://fhir.nl/fhir/NamingSystem/uzi-nr-pers"/>
<value value="006797896"/>
</identifier>
<!--naamgegevens-->
<name>
<family>
<extension url="http://hl7.org/fhir/StructureDefinition/humanname-own-prefix">
<valueString value="van"/>
</extension>
<extension url="http://hl7.org/fhir/StructureDefinition/humanname-own-name">
<valueString value="Beek"/>
</extension>
</family>
<given value="Johan">
<extension url="http://hl7.org/fhir/StructureDefinition/iso21090-EN-qualifier">
<valueCode value="BR"/>
</extension>
</given>
<given value="J.">
<extension url="http://hl7.org/fhir/StructureDefinition/iso21090-EN-qualifier">
<valueCode value="IN"/>
</extension>
</given>
</name>
</Practitioner>
</contained>
<!--voorschrijver / zorgaanbieder-->
<contained>
<Organization>
<id value="d3e83"/>
<identifier>
<system value="http://fhir.nl/fhir/NamingSystem/ura"/>
<value value="00765332"/>
</identifier>
<identifier>
<system value="http://fhir.nl/fhir/NamingSystem/agb-z"/>
<value value="01028432"/>
</identifier>
<name value="Huisartspraktijk Van Beek"/>
</Organization>
</contained>
<extension url="http://nictiz.nl/fhir/StructureDefinition/zib-MedicationUse-Duration">
<valueDuration>
<value value="10"/>
<unit value="dag"/>
<system value="http://unitsofmeasure.org"/>
<code value="d"/>
</valueDuration>
</extension>
<extension url="http://nictiz.nl/fhir/StructureDefinition/zib-Medication-MedicationTreatment">
<valueIdentifier>
<system value="urn:oid:1.2.3.999"/>
<value value="medbeh_VMA_001"/>
</valueIdentifier>
</extension>
<identifier>
<system value="urn:oid:1.2.3.999"/>
<value value="medbeh_VMA_001_MA02"/>
</identifier>
<intent value="order"/>
<category><!--medicatieafspraak-->
<coding>
<system value="http://snomed.info/sct"/>
<code value="16076005"/>
<display value="Prescription (procedure)"/>
</coding>
</category>
<!--geneesmiddel-->
<medicationCodeableConcept>
<coding>
<system value="urn:oid:12.3.999"/>
<code value="19844"/>
<display value="19844 DOXYCYCLINE TABLET 100MG"/>
</coding>
</medicationCodeableConcept>
<subject>
<identifier>
<system value="http://fhir.nl/fhir/NamingSystem/bsn"/>
<value value="32456487"/>
</identifier>
<display value="Antje de Vries"/>
</subject>
<!--afspraakdatum-->
<authoredOn value="2016-09-02T14:30:23+02:00"/>
<requester>
<agent>
<reference value="#d3e58"/>
<display value="Johan J. van Beek"/>
</agent>
<onBehalfOf>
<reference value="#d3e83"/>
<display value="Huisartspraktijk Van Beek"/>
</onBehalfOf>
</requester>
<!--reden afspraak-->
<reasonCode>
<coding>
<system value="http://snomed.info/sct"/>
<code value="438833006"/>
<display value="Contra-indicatie"/>
</coding>
</reasonCode>
<!--toelichting-->
<note>
<text value="Patiënt gebruikt Verapamil, contra-indicatie voor het origineel voorgeschreven Klacid."/>
</note>
<dosageInstruction>
<sequence value="1"/>
<text value="Gedurende 10 dagen, eerste dag 2 tabletten en vervolgens 1x daags 1 tablet"/>
<timing>
<repeat><!--doseerduur-->
<boundsDuration>
<value value="1"/>
<unit value="dag"/>
<system value="http://unitsofmeasure.org"/>
<code value="d"/>
</boundsDuration>
<!--toedieningsduur-->
<frequency value="1"/>
<!--frequentie/tijdseenheid-->
<period value="1"/>
<periodUnit value="d"/>
</repeat>
</timing>
<route>
<extension url="http://hl7.org/fhir/StructureDefinition/iso21090-nullFlavor">
<valueCode value="NI"/>
</extension>
</route>
<doseQuantity>
<value value="2"/>
</doseQuantity>
</dosageInstruction>
<dosageInstruction>
<sequence value="2"/>
<text value="Gedurende 10 dagen, eerste dag 2 tabletten en vervolgens 1x daags 1 tablet"/>
<timing>
<repeat><!--doseerduur-->
<boundsDuration>
<value value="9"/>
<unit value="dag"/>
<system value="http://unitsofmeasure.org"/>
<code value="d"/>
</boundsDuration>
<!--toedieningsduur-->
<frequency value="1"/>
<!--frequentie/tijdseenheid-->
<period value="1"/>
<periodUnit value="d"/>
</repeat>
</timing>
<route>
<extension url="http://hl7.org/fhir/StructureDefinition/iso21090-nullFlavor">
<valueCode value="NI"/>
</extension>
</route>
<doseQuantity>
<value value="1"/>
</doseQuantity>
</dosageInstruction>
<!-- Responsible dispensing pharmacy -->
</MedicationRequest>
</resource>
</entry>
</Bundle>
```

**Comment by ewoutkramer** [comment]
This is what the Java validator does:

Grahame Grieve: depends on the binding strength
Grahame Grieve: if it's **required** those are both an error

Grahame Grieve: if it's **extensible**, it's an error if there's no text

Grahame Grieve: **otherwise** it's not an error

**Comment by ewoutkramer** [comment]
So, we will need to adapt the validator to use the same approach.

**Comment by ewoutkramer** [comment]
Fixed by #931.
